### PR TITLE
MOB-3886

### PR DESCRIPTION
**Pull request by mateid-kr**
## JIRA
[MOB-3886](https://karhoo.atlassian.net/browse/MOB-3886)

## DESCRIPTION 
Improve loyalty component UX when for errors : standard, insufficient balance, currency not supported

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [X] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Review by mohadian — COMMENTED** [review]
The internal-error need to be presented to the user, so they would be aware that it is an interim error and they can try again later.
The kh_uisdk_loyalty_points_earned_for_trip needs to be removed from the layout so it wont display the `%1$s` in the UI

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutView.kt [hunk 1]
```diff
@@ -112,6 +114,24 @@ internal class CheckoutView @JvmOverloads constructor(context: Context,
             }
         }
 
+        loyaltyView.delegate = object : LoyaltyContract.LoyaltyViewDelegate {
+            override fun onModeChanged(mode: LoyaltyMode) {
+                if (loyaltyView.getCurrentMode() == LoyaltyMode.ERROR_BAD_CURRENCY) {
+                    loadingButtonCallback.enableButton(false)
```
Comment (dianakarh) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutView.kt:120]: What if the user wants to book a ride but is uninterested if the loyalty component is working. IMHO the errors should prevent the user from burning, not booking (with or without earning where possible)

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/loyalty/LoyaltyPresenter.kt [hunk 2]
```diff
@@ -158,6 +182,8 @@ class LoyaltyPresenter(val userStore: UserStore = KarhooApi.userStore,
             loyaltyService.getLoyaltyStatus(loyaltyId).execute { result ->
                 when (result) {
                     is Resource.Success -> {
+//                        val loyaltyStatus = LoyaltyStatus(1500, canBurn = true, canEarn = true)
```
Comment (dianakarh) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/loyalty/LoyaltyPresenter.kt:185]: Is this necessary for testing or just a left over from development?

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutView.kt [hunk 3]
```diff
@@ -466,8 +457,33 @@ internal class CheckoutView @JvmOverloads constructor(context: Context,
     override fun isPaymentMethodValid(): Boolean = bookingRequestPaymentDetailsWidget.hasValidPaymentType()
 
     override fun checkLoyaltyEligiblityAndStartPreAuth(): Boolean {
-        return if (loyaltyView.visibility == VISIBLE && loyaltyView.getCurrentMode() != LoyaltyMode.ERROR) {
-            loyaltyView.preAuthorize()
+        return if (loyaltyView.visibility == VISIBLE &&
+            loyaltyView.getCurrentMode() == LoyaltyMode.ERROR_BAD_CURRENCY) {
```
Comment (dianakarh) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/component/views/CheckoutView.kt:461]: Shouldn't this be `!=`?

**Review by mohadian — APPROVED** [review]

**Review by dianakarh — APPROVED** [review]

**Pull request merged**